tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
+
+				if ruleName != "" {
+					preRuleNameList = append(preRuleNameList, ruleName)
+				}
+			}
+			log.InfoD("Creating post rule for deployed apps")
+			for i := 0; i < len(appList); i++ {
+				postRuleStatus, ruleName, err := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "post")
+				log.FailOnError(err, "Creating post rule for deployed apps failed")
+				dash.VerifyFatal(postRuleStatus, true, "Verifying Post rule for backup")
+				if ruleName != "" {
+					postRuleNameList = append(postRuleNameList, ruleName)
+				}
+			}
+		})
+
+		Step("Creating bucket,backup location and cloud setting", func() {
+			log.InfoD("Creating bucket,backup location and cloud setting")
+			for _, provider := range providers {
+				cloudCredName = fmt.Sprintf("%s-%s-%v", "cred", provider, time.Now().Unix())
+				bkpLocationName = fmt.Sprintf("%s-%s-bl", provider, getGlobalBucketName(provider))
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = bkpLocationName
+				CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID)
+				err := CreateBackupLocation(provider, bkpLocationName, backupLocationUID, cloudCredName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, "Creating backup location")
+			}
+		})
+
+		// Step("Register cluster for backup" is not needed as it has been executed as psrt of K8s version validation
+
+		Step("Taking backup of application fron source cluster", func() {
+			log.InfoD("Taking backup of applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting context")
+			backupNames = make([]string, len(sourceNamespaces))
+			backupContexts = make([]*BackupRestoreContext, len(sourceNamespaces))
+			for i, namespace := range sourceNamespaces {
+				backupName := fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				backupNames[i] = backupName
+				backupCtx, err := CreateBackupAndGetBackupCtx(backupName, SourceClusterName, bkpLocationName, backupLocationUID, []string{namespace}, labelSelectors, orgID, clusterUid, "", "", "", "", ctx, []*scheduler.Context{sourceClusterAppsContexts[i]})
+				dash.VerifyFatal(err, nil, "Verifying backup creation")
+				backupContexts[i] = backupCtx
+			}
+		})
+
+		Step("Restoring the backed up applications on destination cluster", func() {
+
+			log.InfoD("Restoring the backed up applications on destination cluster")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			for i, sourceNamespace := range sourceNamespaces {
+				var initialRestoreName string
+
+				Step("Restoring the backed up application to namespace of same name on destination cluster", func() {
+					log.InfoD("Restoring the backed up application to namespace of same name on destination cluster")
+
+					initialRestoreName = fmt.Sprintf("%s-%s-initial-%v", restoreNamePrefix, sourceNamespace, time.Now().Unix())
```

Comment:
Please do variable assignment using `:=` instead of declaring it as `var` and then assigning it

Reply:
I cannot do that as the same variable is used elsewhere in another `step` and initializing it inside the step will limit the scope.

Reply:
✅